### ISSUE TITLE
Tabstops automation with modal dialogs

### ISSUE THREAD

**Issue by ferBonnin** [microsoft/accessibility-insights-web#5305]
## Describe the bug

If there is a modal dialog and you run Tabstops, all elements behind the modal dialog will be marked as failures

**To Reproduce**

1. Go to https://www.washington.edu/accesscomputing/AU/after.html
2. Click on the cheat sheet of a11y issues
3. Run tabstops over the modal dialog
4. See that all other elements in the page are marked as potential failures

## Expected behavior
Only the interactable elements should be marked as potential failures

## Screenshots

![a modal dialog with tabstops, all the elements on the back are marked as potential failures](https://user-images.githubusercontent.com/9062104/160463419-a19f91ee-c4e4-4d8b-af47-9c763dacfeff.png)

**Comment by ferBonnin** [comment]
triaged with Peter. Even though issues are added as potential / needs review, this could confuse users and make them not keep the focus in the modal only

**Comment by sfoslund** [comment]
I think the root cause of this is an oddity in tabbable- see its [readme](https://github.com/focus-trap/tabbable#more-details) which states: `Only elements hidden with display: none or visibility: hidden are left out`. It looks like it ignores `aria-hidden=true` and includes those elements in its list of tabbable elements, which causes us to mark them as missing when the user is unable to tab to them.

@waabid were you aware of this behavior previously? Do you think it's something we can file a bug on them for or something we'll have to adapt to from our end?

**Comment by waabid** [comment]
Unfortunately, this is news to me as well. I guess we'll have to self-filter after getting the list of elements. Filtering for components that are under aria-hidden (and aria-modal, per WCAG 1.1) should be doable. Still, I do think there might be other modals (usually ones that are not accessible per WCAG since my understanding is one of aria-hidden or aria-modal attributes are required) that exist that can cause this issue.

Wonder if it makes sense to maybe add a note for this scenario?

**Comment by sfoslund** [comment]
@waabid makes sense, I made a PR to do that filtering: https://github.com/microsoft/accessibility-insights-web/pull/5308

**Comment by sfoslund** [comment]
I think there's also a false positive keyboard trap on this page- notice the error on tab # 4 over the close button in the screenshot above. Investigating initially it looks like the [oldElement here](https://github.com/microsoft/accessibility-insights-web/blob/main/src/injected/analyzers/tab-stops-orchestrator.ts#L108) is incorrect when tabbing from the OK to the close button- the old element should be the OK button but instead it's the close button. I'm not sure why we're seeing this on this particular site, I can't find anything odd about it so far.

**Comment by sfoslund** [comment]
This issue was fixed by #5308, but that PR did not address the false positive focus trap I mentioned above so I opened https://github.com/microsoft/accessibility-insights-web/issues/5317 to track that issue separately.

**Comment by sfoslund** [comment]
Closing as this was fixed in #5308
